### DISCUSSION TITLE
Resolve the "TransactionScope - 'Transaction management request 0 not supported'" error.

### DISCUSSION THREAD

**Discussion by amalbabu-94** [babelfish-for-postgresql/babelfish-for-postgresql#36] [Discussion]
We get this error   "TransactionScope - 'Transaction management request 0 not supported'"  whenever we try to call the below code while using babelfish.We have made sure the isolation levels which we use are supported by Babelfish.Any idea what can be the possible cause of this exception here?
try
{
    mTransactionScope = new TransactionScope(TransactionScopeOption.Required, transactionOptions);
}
catch (Exception)
{}

**Comment by staticlibs** [comment]
@amalbabu-94 

When the `TransactionScope` is active (when it is using ambient transaction context), it sends the following to the DB server to request the address of the Distributed Transaction Coordinator (DTC):

```
Tabular Data Stream
    Type: Transaction Manager Request (14)
    Status: 0x01, End of message
    Length: 34
    Channel: 0
    Packet Number: 1
    Window: 0
    Transaction Manager Request Packet
        Packet data stream headers
            Total length: 22
            Header
                Length: 18
                Type: Transaction descriptor (0x0002)
                Transaction descriptor: 0
                Outstanding request count: 1
        Transaction Manager Request: TM_GET_DTC_ADDRESS (0)
        Payload
```

DTC is not supported in Babelfish (it is not guaranteed to be enabled in MSSQL either), and because the value of `TM_GET_DTC_ADDRESS` option is `0` - the error `request 0 not supported` is returned.

It seems to be not possible to force the `TransactionScope` to use only local transactions. So it cannot be used with Babelfish, explicit transaction handling with `System.Transactions` can be used instead.
